File: 2026-09/src/day_04/leet_198/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(1 + n);

        for (int i = 1; i <= n; i++) {
            dp[i] = max(dp[i - 1], dp[max(i - 2, 0)] + nums[i - 1]);
        }

        return dp[n];
    }
};

File: 2026-09/src/day_04/leet_198/Solution.java
package day_04.leet_198;

class Solution {
    public int rob(int[] nums) {
        int n = nums.length;
        int[] dp = new int[1 + n];

        for (int i = 1; i <= n; i++) {
            dp[i] = Math.max(dp[i - 1], dp[Math.max(i - 2, 0)] + nums[i - 1]);
        }

        return dp[n];
    }
}
